src/compiler.rkt

```diff
@@ -26,23 +27,30 @@
 (define (make-progs-interpreter name vars ivec roots)
   (define rootvec (list->vector roots))
   (define rootlen (vector-length rootvec))
+  (define iveclen (vector-length ivec))
   (define varc (length vars))
-  (define vreg-count (+ varc (vector-length ivec)))
+  (define vreg-count (+ varc iveclen))
   (define vregs (make-vector vreg-count))
-  (define vprecs (make-vector (vector-length ivec)))   ; vector that stores working precisions
+  (define vrepeats (make-vector iveclen #f))           ; flags whether an op should be evaluated
+  (define vprecs (make-vector iveclen))                ; vector that stores working precisions
   (define vstart-precs (setup-vstart-precs ivec varc)) ; starting precisions for the tuning mode
+  
   (define prec-threshold (/ (*max-mpfr-prec*) 25))     ; parameter for sampling histogram table
   (if (equal? name 'ival)
       (λ args
         (define timeline-stop! (timeline-start!/unsafe 'mixsample "backward-pass"
                                                        (* (*sampling-iteration*) 1000)))
-        (when (not (zero? (*sampling-iteration*)))                  
-          (backward-pass ivec varc vregs vprecs vstart-precs rootvec)) ; back-pass
+        (if (zero? (*sampling-iteration*))
+            (vector-fill! vrepeats #f)
```

Comment:
I suppose we could have a hard-coded vector of `#f` and pull the same trick as with `vprecs` right?